### DISCUSSION TITLE
Meaning and cause of "Restore will override" error

### DISCUSSION THREAD

**Discussion by Huntsekkern** [WalletConnect/walletconnect-monorepo#3098] [Developer Support]
Hello, 
I'm working on a modified variant of the EthereumProvider and the UniversalProvider of the walletconnect-monorepo.

While testing with existing walletconnect tests, I often get the error `Restore will override. subscription`, during the set-up phase and subsequently, I tend to receive a lot of `irn_subscription` from the staged relay server until the test times out.

I am not able to pinpoint what this error means, nor what nor where in the code does it come from. Even just understanding what is being restored and what is being override would be helpful!

I get something like that:
```
{"level":50,"time":1689083697474,"pid":523090,"hostname":"localhost","context":"core","context":"core/relayer","context":"core/relayer/subscription","msg":"subscription: [{\"topic\":\"675172a7dbb1475b779d5988ac4868be58743641a872087e4fc8a43d1545b38d\",\"relay\":{\"protocol\":\"irn\"},\"id\":\"029409b5cdfb06801f3f5a3947561a0810ce7a936851b90196beb255b201d9d7\"}]"}
{"level":50,"time":1689083697474,"pid":523090,"hostname":"localhost","context":"core","context":"core/relayer","context":"core/relayer/subscription","err":{"type":"Error","message":"Restore will override. subscription","stack":"Error: Restore will override. subscription\n    at tt.restore (/root/walletconnect-monorepo-nym/packages/core/dist/index.cjs.js:1:30185)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async tt.restart (/root/walletconnect-monorepo-nym/packages/core/dist/index.cjs.js:1:25608)\n    at async tt.onConnect (/root/walletconnect-monorepo-nym/packages/core/dist/index.cjs.js:1:30670)\n    at async EventEmitter.<anonymous> (/root/walletconnect-monorepo-nym/packages/core/dist/index.cjs.js:1:31042)"},"msg":"Restore will override. subscription"}
```

Thank you for your input!

**Comment by crypblizz8** [comment]
@ganchoradkov

**Comment by Huntsekkern** [comment]
While I'm not fully at clear with it still, I solved other issues and I do not encounter this error anymore.
It comes from some files in core/controllers, such as store.ts, history.ts, expirer.ts when the function restore is called.
